modules/azure/event_grid_topic_subscription/variables.tf

```diff
@@ -24,3 +24,16 @@ variable "service_bus_topic_id" {
   description = "Service Bus topic ID where the message should be forwarded"
   default     = null
 }
+variable "subject_filter" {
```

Comment:
Is subject filter itself also optional (since all it's underlying var's are)? If it is, it might be better to mark it dynamic in the main and default it to null rather then no default on the var and all underlying objects to be optional. I'm not entirely sure, but currently I think you would have to submit an empty subject_filter = {} since the param itself is not defaulted. If that's not the case it doesn't matter much.

Reply:
Indeed, Made it Dynamic